modules/nf-core/bamixchecker/main.nf

```diff
@@ -0,0 +1,62 @@
+process BAMIXCHECKER {
+    tag "$meta.id"
+    label 'process_low'
+
+    // conda "${moduleDir}/environment.yml" //not in conda
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker.io/abenjak/bamixchecker:1.0.1' :
+        'docker.io/abenjak/bamixchecker:1.0.1' }"
+
+
+    input:
+    tuple val(meta), path(bams)
+    tuple val(meta2), path(fasta)
+    tuple val(meta2), path(fai)
+    tuple val(meta2), path(dict)
+    tuple val(meta3), path(target_bed) //optional via non-empty file
+    tuple val(meta4), path(nhSNP)  //optional via non-empty file
+
+    output:
+    tuple val(meta), path("BAMixChecker/*.pdf")            , emit: heatmap
+    tuple val(meta), path("BAMixChecker/Total_result.txt") , emit: total_result
+    tuple val(meta), path("BAMixChecker/*.html")           , emit: html
+    tuple val(meta), path("BAMixChecker/*_samples.txt")    , emit: results
+    path  "versions.yml"                                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def VERSION = '1.0.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    """
+    python /BAMixChecker-1.0.1/BAMixChecker.py \\
```

Comment:
This path isn't correct on the CI, you need to find where it located inside the images.

Reply:
Thanks for your comment.
I don't understand, this is the correct full path inside the image.
Pulling a singularity image from `docker.io/abenjak/bamixchecker:1.0.1`, then:
```
singularity exec BAMixChecker_v1.0.1.sif python /BAMixChecker-1.0.1/BAMixChecker.py -h
usage: BAMixChecker [-h] [-d DIR] [-l LIST] -r REF [-o OUTPUTDIR] [-b BEDFILE]
                    [-v {hg38,hg19}] [-p MAXPROCESS] [-nhSNP NONHUMANSNPLIST]
                    [-pld PLOIDY] [--FullPATH] [--RemoveVCF]
                    [--OFFFileNameMatching]
```
How else can I run `BAMixChecker.py` without providing the full path in the image and excluding the root `/`?
(A more elegant solution would have been to put `BAMixChecker.py` in the PATH inside the image, but does it matter?)
Best

Reply:
Oh, it is on conda that it is failing with:
    >   python: can't open file '/BAMixChecker-1.0.1/BAMixChecker.py': [Errno 2] No such file or directory
because you have specified python 3 and nothing else in the environment.yml.
We much prefer to have tools distributed via bioconda, as a package there, rather than custom docker containers; that will automatically make the tool available on docker and singularity via the bioconductors project.

Reply:
I see. Yeah this was my workaround because the tool is not on bioconda. Would you prefer that I add it to bioconda? (I have no experience with bioconda, can anyone easily add any tool there?)

Reply:
Anyone can add tools to bioconda yes. Easily is another matter ;)

Reply:
I don't have much experience in contributing to bioconda.